streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -318,6 +320,16 @@ public void foreach(ForeachAction<? super K, ? super V> action) {
     }
 
     @Override
+    public KStream<K, V> peek(final ForeachAction<? super K, ? super V> action) {
```

Comment:
You should add a test to `KStreamImplTest`: `shouldNotAllowNullActionOnPeek` or similar.